packages/astro/src/integrations/index.ts

```diff
@@ -136,6 +136,14 @@ export async function runHookConfigSetup({
 					}
 					addedClientDirectives.set(name, buildClientDirectiveEntrypoint(name, entrypoint));
 				},
+				addMiddleware: ({ order, entrypoint }) => {
+					if(typeof updatedSettings.middleware[order] === 'undefined') {
+						throw new Error(
+							`The "${integration.name}" integration is trying to add middleware but did not specify an order.`
+						);
+					}
+					updatedSettings.middleware[order].push(entrypoint);
```

Comment:
Can we have a `logger.debug` here saying that "Integration X added a middleware on <ORDER> position"?

Reply:
Ah yes, I had put it in the virtual module code, but this is probably a better location as we know the integration's name.